cpp/tests/datetime/datetime_ops_test.cpp

```diff
@@ -382,4 +391,139 @@ TEST_F(BasicDatetimeOpsTest, TestDayOfYearWithEmptyColumn)
   EXPECT_EQ(out_col->size(), 0);
 }
 
+TEST_F(BasicDatetimeOpsTest, TestAddMonthsWithInvalidColType)
+{
+  using namespace cudf::test;
+  using namespace cudf::datetime;
+  using namespace simt::std::chrono;
+
+  // Time in seconds since epoch
+  // Dates converted using epochconverter.com
+  auto timestamps_s =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_s, cudf::timestamp_s::rep>{
+      662688000L  // 1991-01-01 00:00:00 GMT
+    };
+
+  // Months has to be an INT16 type
+  auto months = cudf::test::fixed_width_column_wrapper<int32_t>{-2};
+
+  EXPECT_THROW(add_calendrical_months(timestamps_s, months), cudf::logic_error);
+}
+
+TEST_F(BasicDatetimeOpsTest, TestAddMonthsWithIncorrectColSizes)
+{
+  using namespace cudf::test;
+  using namespace cudf::datetime;
+  using namespace simt::std::chrono;
+
+  // Time in seconds since epoch
+  // Dates converted using epochconverter.com
+  auto timestamps_s =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_s, cudf::timestamp_s::rep>{
+      662688000L  // 1991-01-01 00:00:00 GMT
+    };
+
+  // Provide more number of months rows than timestamp rows
+  auto months = cudf::test::fixed_width_column_wrapper<int16_t>{-2, 3};
+
+  EXPECT_THROW(add_calendrical_months(timestamps_s, months), cudf::logic_error);
+}
+
+TEST_F(BasicDatetimeOpsTest, TestAddMonthsWithSeconds)
+{
+  using namespace cudf::test;
+  using namespace cudf::datetime;
+  using namespace simt::std::chrono;
+
+  // Time in seconds since epoch
+  // Dates converted using epochconverter.com
+  auto timestamps_s =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_s, cudf::timestamp_s::rep>{
+      662688000L,   // 1991-01-01 00:00:00 GMT
+      949496401L,   // 2000-02-02 13:00:01 GMT - leap year
+      1056931201L,  // 2003-06-30 00:00:01 GMT - last day of month
+      1056964201L,  // 2003-06-30 09:10:01 GMT - last day of month
+      1056974401L,  // 2003-06-30 12:00:01 GMT - last day of month
+      1056994021L,  // 2003-06-30 17:27:01 GMT - last day of month
+      0L,           // This is the UNIX epoch - 1970-01-01
+      0L,           // This is the UNIX epoch - 1970-01-01
+      -131586588L,  // 1965-10-31 00:10:12 GMT
+      -131550590L,  // 1965-10-31 10:10:10 GMT
+      -131544000L,  // 1965-10-31 12:00:00 GMT
+      -131536728L   // 1965-10-31 14:01:12 GMT
+    };
+
+  auto months =
+    cudf::test::fixed_width_column_wrapper<int16_t>{-2, 6, -1, 1, -4, 8, -2, 10, 4, -20, 1, 3};
+
+  expect_columns_equal(
```

Comment:
`CUDF_TEST_EXPECT_COLUMNS_EQUAL` macro makes easier to point out line of failure.

Reply:
done